python/salpytools/salpylib.py

```diff
@@ -440,23 +444,50 @@ def getCurrentEvent(self):
     def getCurrentCommand(self):
         return self.getCurrent()
 
-    def waitEvent(self, tsleep=None, timeout=None):
-        """ Loop for waiting for new event """
+    def waitEvent(self, tsleep=None, timeout=None, after_timeStamp=-1):
+        """
+        Loops while waiting for a new event.
+
+        This function waits until the class gets a new event via the
+        variable self.newEvent. The self.newEvent is crontrolled by the run_Event function.
+        - Once the self.newEvent changes to True, the function makes sure that the timeStamp
+        in this new event happened AFTER the after_timeStamp time set in the function call. This
+        ensures that no rogue newEvent trigger the wait. If this is a bona-fide new Event, then we
+        break from the loop. By default after_timeStamp=-1, to make sure dtime is positive in case
+        that after_timeStamp is not defined.
+        - If the time inside the wait loop exceeds the timemout set time, them the function breaks
+        from the loop and returns self.timeoutEvent=True and self.newEvent=False
+        """
         if not tsleep:
             tsleep = self.tsleep
         if not timeout:
             timeout = self.timeout
 
         t0 = time.time()
-        while not self.newEvent:
+        self.timeoutEvent = False
+
+        while True:
             sys.stdout.flush()
-            sys.stdout.write("Wating for %s event.. [%s]" % (self.topic, next(spinner)))
+            sys.stdout.write("Waiting for {} event.. [{}]".format(self.topic, next(spinner)))
             sys.stdout.write('\r')
-            if time.time() - t0 > timeout:
-                LOGGER.warning("Timeout reading for Event %s" % self.topic)
+            if self.newEvent:
+                # Make sure that self.newEvent happens AFTER the time stamp requested time
+                dtime = self.timeStamp - after_timeStamp
```

Comment:
If `self.timeStamp` is not set (as can happen with the change above), this will fail.  Should there be some kind of guard against that?

Reply:
Added new function to better treat rogue events